test/Adapter/PhpCodeTest.php

```diff
@@ -26,111 +25,71 @@ public function setUp()
         $this->adapter = new Serializer\Adapter\PhpCode();
     }
 
-    public function tearDown()
-    {
-        $this->adapter = null;
-    }
-
-    public function testSerializeString()
-    {
-        $value    = 'test';
-        $expected = "'test'";
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testSerializeFalse()
-    {
-        $value    = false;
-        $expected = 'false';
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testSerializeNull()
-    {
-        $value    = null;
-        $expected = 'NULL';
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
-    public function testSerializeNumeric()
-    {
-        $value    = 100.12345;
-        $expected = '100.12345';
-
-        $data = $this->adapter->serialize($value);
-        $this->assertEquals($expected, $data);
-    }
-
+    /**
+     * Test when serializing a PHP object it matches the
+     * encode process
+     *
+     * Unserialize on PHP objects occur on Zend\Json\Encoder::encode
+     */
     public function testSerializeObject()
     {
-        $value    = new \stdClass();
```

Comment:
This test should work without changes. Otherwise this PR introduces a BC Break.

You'll find why JSON Encoder is not the solution. The adapter emulate php native serialize/unserialize without call the native implementation.

JSON Encoded adapter is this https://github.com/zendframework/zend-serializer/blob/master/src/Adapter/Json.php